rma/wizard/rma_delivery.py

```diff
@@ -66,14 +66,9 @@ def default_get(self, fields_list):
 
     @api.onchange("product_id")
     def _onchange_product_id(self):
-        domain_product_uom = []
         if self.product_id:
-            domain_product_uom = [
-                ("category_id", "=", self.product_id.uom_id.category_id.id)
-            ]
             if not self.product_uom or self.product_id.uom_id.id != self.product_uom.id:
                 self.product_uom = self.product_id.uom_id
-        return {"domain": {"product_uom": domain_product_uom}}
```

Comment:
Why do you think this should be removed?

Reply:
In the Odoo logs, I had this warning:
....onchange method RmaReDeliveryWizard._onchange_product_id returned a domain, this is deprecated

Reply:
Solution1:
1.add a field `uom_category_id = fields.Many2one(related='product_id.uom_id.category_id')`
2. and add on view `domain="[('category_id', '=', uom_category_id)]"` for `product_uom`
Solution2:
1.add a field `allowed_uom_ids = fields.Many2many('uom.uom', compute="_compute_allowed_uoms")`
    @api.depends('product_id')
    def _compute_allowed_uoms(self):
        ...
2. and add on view `domain="[('id '=', allowed_uom_ids)]"` for `product_uom`